llvm/tools/llvm-ir2vec/llvm-ir2vec.cpp

```diff
@@ -283,9 +335,32 @@ int main(int argc, char **argv) {
       "See https://llvm.org/docs/CommandGuide/llvm-ir2vec.html for more "
       "information.\n");
 
+  // Validate input file requirement
+  if (InputFilename.empty() && Mode != EntityMode) {
+    errs() << "Error: Input file (.bc/.ll) or stdin (-) is required\n";
```

Comment:
You should probably have a test for this case.

Reply:
Seems that this check is not needed. When input files are not provided it "just waits" for input (like other llvm tools). So removed the check.